Add hascuSOLVER flag to Context

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70253
* #69827
* #69826
* __->__ #69825

As per title.

cc @ngimel @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

Differential Revision: [D33751986](https://our.internmc.facebook.com/intern/diff/D33751986)

lezcano committed:
Add hascuSOLVER flag to Context

As per title.

[ghstack-poisoned]

IvanYashchuk commented:
Maybe we should also expose `CUSOLVER_VERSION` and `MAGMA_VERSION_MAJOR, MAGMA_VERSION_MINOR, MAGMA_VERSION_MICRO`...

Lezcano commented:
I'll wait for the CI to pass. If it does, I'll add those things to this PR.

mruberry commented on aten/src/ATen/Context.h:
```diff
@@ -74,6 +74,9 @@ class TORCH_API Context {
   static long versionCuDNN() {
     return detail::getCUDAHooks().versionCuDNN();
   }
+  static bool hascuSOLVER() {
```
I get this should be `hasCuSOLVER` to be consistent with `versionCuDNN` above?

mruberry reviewed (approved):
Seems OK to be but @ngimel is our CUDAHooks expert

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

Commit:
fixes typo in function name

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33202687).

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

lezcano committed:
Update on "Add hascuSOLVER flag to Context"


As per title.

cc ngimel jianyuh nikitaved pearu mruberry walterddr IvanYashchuk xwang233 Lezcano

Differential Revision: [D33202687](https://our.internmc.facebook.com/intern/diff/D33202687)

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33751986).
